### DISCUSSION TITLE
./run.sh designer error - javafx missing

### DISCUSSION THREAD

**Discussion by niuzhi** [pmd/pmd#2565] [Q&A] [answered]
Hello，my java version=1.4，javafx=1.3.1_b101,pmd=6.24.0
Run command “./run.sh designer” produces the following exception，how to slove?
![image](https://user-images.githubusercontent.com/33443724/83725413-7a825380-a674-11ea-9311-4264544f8ef6.png)

**Comment by adangel** [comment, marked as answer]
Hi @niuzhi, you need to use at least javafx 11 (you are using the 10 year old javafx 1.3).
JavaFX is available from https://gluonhq.com/products/javafx/

After installing/extracting JavaFX, you'll need to set the environment variable `JAVAFX_HOME` to the directory, where you extracted JavaFX.

See also https://pmd.github.io/latest/pmd_userdocs_extending_designer_reference.html#installing-running-updating

Alternatively, you can use Oracle Java 8, which has javafx included.
